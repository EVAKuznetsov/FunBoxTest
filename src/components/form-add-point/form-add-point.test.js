import React from 'react'
import FormAddPoint from './form-add-point'

describe('Form add point',()=>{ 
  const props = {
    addPoint:jest.fn()
  }
  const formAddPoint = shallow(<FormAddPoint {...props} />);
  it('renders form',()=>{
    expect(shallowToJson(formAddPoint)).toMatchSnapshot()
  })
  it('isForm',()=>{
    expect(formAddPoint.find('form')).toHaveLength(1)
  })
  it('entry in state',()=>{
    const input = formAddPoint.find('input');          
    input.simulate('change', { target: { value: 'test text' } });
    expect(formAddPoint.state().label).toEqual('test text');
  })
  it('submit',()=>{
    const form = formAddPoint.find('form');
    const input = formAddPoint.find('input');          
    input.simulate('change', { target: { value: 'test text' } });
    form.simulate('submit',{ preventDefault() {} });
    expect(props.addPoint).toBeCalled();
    expect(formAddPoint.state().label).toEqual('');
  })  
})