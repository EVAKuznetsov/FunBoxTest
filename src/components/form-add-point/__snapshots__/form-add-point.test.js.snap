// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form add point renders form 1`] = `
<form
  className="form-group"
  onSubmit={[Function]}
>
  <input
    className="form-control"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
